python/tach/parsing/modules.py

```diff
@@ -44,13 +45,21 @@ def find_cycles(
         for dependency in module.depends_on:
             graph.add_edge(module.path, dependency.path)  # type: ignore
 
-    all_cycles: list[list[str]] = list(nx.simple_cycles(graph))  # type: ignore
-
-    canonical_cycles = {tuple(canonical_form(cycle)) for cycle in all_cycles}
-
-    unique_cycles = [list(cycle) for cycle in canonical_cycles]
+    modules_with_cycles: list[str] = []
+    for module in modules:
+        module_path = module.path
+        try:
+            # Find *any* cycle, starting with module_path
+            cycle: list[tuple[str, str]] = nx.find_cycle(graph, source=module_path)  # type: ignore
```

Comment:
`source` is confusing in `find_cycle` - it just starts with `source`, but moves on to other nodes if there's no cycle found with `source`. 

There's not a better method to call either unfortunately, so we need to check that the cycle found contains `source` after the fact. It does at least mean if we don't find any cycles on the first call we can just return.